### PR TITLE
DM-27753: exposurelog: add the ability to mount file systems

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: charts/exposurelog/templates/configmap-fs-mounts.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
```
Comment (afausti) [inline charts/exposurelog/templates/configmap-fs-mounts.yml:1]: That's correct, apiVersion here is for the kubernetes api.

**Review by afausti — COMMENTED** [review]
- If you don't need any mount point in your deployment you can control that with a flag similar to here: https://github.com/lsst-sqre/charts/blob/master/charts/nublado/templates/jupyterhub/deployment.template.yml#L191 

- That's also true for the entire configmap.  Something similar to this https://github.com/lsst-sqre/charts/blob/master/charts/nublado/templates/jupyterhub/vault-secrets.template.yml#L1 I think it is fine using a configmap to specify the mount options.

- There's something missing in the deployment template. You need a `volumeMount` in the containter specification https://kubernetes.io/docs/concepts/storage/volumes/#configmap

**Comment by r-owen** [comment]
I decided to just make the default to have a list of 0 mount points and put that default in values.yaml. That way the files in lsp-deploy need only specify mountpoints if they have one (or several).

I will say that this service will probably only ever have one or zero mountpoints, but it seems more work than it's worth to switch to allowing just one optional mountpoint, instead of a list of 0 or more items. I think it's right now -- at least it passes git actions and I can render the template locally (with the values from lsp-deploy for nts and with the default of no mountpoints). Whether the rendered data is correct I'm afraid I have no idea. I was hoping to see the specified mountpoint data in more than just one location.

Here is the result from rendering the example (NTS example):
```
# Source: exposurelog/templates/configmap-fs-mounts.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: exposurelog-fs-mounts
data:
  mountpoints.json: |
    [
      {
        "hostpath": "/lsstdata/offline/teststand",
        "mountpoint": "/teststand",
        "mode": "r"
      }
    ]
---
...(teststand is not mentioned again)...
---
# Source: exposurelog/templates/deployment.yaml
...
      volumes:
        - name: fs-mounts
          configMap:
            defaultMode: 0444
            items:
            - key: mountpoints.json
              path: mountpoints.json
            name: exposurelog-fs-mounts
---
...
```

File: charts/exposurelog/templates/configmap-fs-mounts.yml (hunk 1 above)
Comment (r-owen) [reply charts/exposurelog/templates/configmap-fs-mounts.yml]: Is that different than the apiVersion in Chart.yaml (which is `v2`)? I thought that was also used as a Kubernetes version and was worried they would be required to match.

Comment (afausti) [reply charts/exposurelog/templates/configmap-fs-mounts.yml]: Yes. `apiVersion` in the Chart.yaml is the Helm API version. The chart is not a Kubernetes API object. In general, Kubernetes objects might have different API groups and versions and they don't need to match. For example, a `ConfigMap` has `apiVersion: v1` a `Deployment` belongs to the `apps` API group and has `apiVersion: apps/v1`, etc

**Comment by afausti** [comment]
Yes, you still need `VolumeMounts` in the container spec in the `deployment.yaml` template. Like in https://github.com/lsst-sqre/charts/blob/master/charts/nublado/templates/jupyterhub/deployment.template.yml#L187

or here:

https://kubernetes.io/docs/concepts/storage/volumes/#configmap

**Review by afausti — COMMENTED** [review]
Looking good. I think the only thing missing is the `VolumeMounts` in deployment.yaml.

**Comment by r-owen** [comment]
I think I have implemented what we discussed. Here is some example output:

Default. I hope it is OK for these keys to exist with no values; I think it'll be treated as a list with 0 elements:
```
          volumeMounts:
      volumes:
```

With hostpath_1 specified:
```
          volumeMounts:
            - mountPath: /volume_1
              name: volume_1
      volumes:
        - name: volume_1
          hostPath:
            path: /lsstdata/offline/teststand
            type: Directory
```

With hostpath_2 specified:
```
         volumeMounts:
            - mountPath: /volume_2
              name: volume_2
      volumes:
        - name: volume_2
          hostPath:
            path: /lsstdata/offline/teststand
            type: Directory
```

**Review by afausti — APPROVED** [review]
That's great, it looks correct to me.  The comment about how to format `butler_uri_1` and `butler_url_2` makes it easy enough to configure.

**Pull request merged**